resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/filesystem/FileStoreTest.java

```diff
@@ -134,25 +135,32 @@ public void testCopyAcrossVolumes() throws Throwable {
 		IFileStore tempSrc = tempDirectories[0];
 		/* get the destination folder */
 		IFileStore tempDest = tempDirectories[1];
+
 		// create tree
-		IFileStore target = tempSrc.getChild("target");
+		String subfolderName = "target_" + System.currentTimeMillis();
+
+		IFileStore target = tempSrc.getChild(subfolderName);
 		createDir(target, true);
 		createTree(getTree(target));
 
 		/* c:\temp\target -> d:\temp\target */
-		IFileStore destination = tempDest.getChild("target");
+		IFileStore destination = tempDest.getChild(subfolderName);
+		deleteOnTearDown(destination);
```

Comment:
why do we still need those deleteOnTearDown here?

Reply:
My is understanding is the following:
If the folder at which the file store root is placed already exists, it will not be deleted on teardown (this is the essential chang ein this PR). Then all child stores have to be deleted on teardown, which is why `deleteOnTearDown` is called for every child.